xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -600,11 +600,26 @@ bool CDVDDemuxFFmpeg::Open(std::shared_ptr<CDVDInputStream> pInput, bool streami
   if (m_checkvideo && m_streaminfo)
   {
     int64_t duration = m_pFormatContext->duration;
+    AVRational frameRate = {0};
+    int streamIndex = -1;
+    for (auto stream : GetStreams())
+    {
+      if (stream->type == StreamType::STREAM_VIDEO)
+      {
+        streamIndex = static_cast<CDemuxStreamVideo*>(stream)->uniqueId;
+        frameRate.num = static_cast<CDemuxStreamVideo*>(stream)->iFpsRate;
+        frameRate.den = static_cast<CDemuxStreamVideo*>(stream)->iFpsScale;
+        break;
+      }
+    }
+
     std::shared_ptr<CDVDInputStream> pInputStream = m_pInput;
     Dispose();
     if (!Open(pInputStream, false))
       return false;
     m_pFormatContext->duration = duration;
+    if (streamIndex >= 0)
+      m_pFormatContext->streams[streamIndex]->r_frame_rate = frameRate;
```

Comment:
That won't work. Video stream might not have been discovered here. I would save framerate and assign it directly to CDemuxStreamVideoFFmpeg in AddStream later.

Reply:
Ok, got you. Thanks.